query-overview.md

```diff
@@ -0,0 +1,122 @@
+| ------------------------------------------------------------------------------------------ | ----------------------------------------------------------------------------------------------------------------------------------------- |
+| Returns one row of data about one entity                                                   | Returns a DataFrame of many rows of historical data corresponding to multiple entities point-in-time                                      |
+| Designed to return data immediately in milliseconds                                        | Blocks until computation is complete, not designed for millisecond-level computation                                                      |
+| Queries the online store, which caches recent data from online queries for quick retrieval | Queries the offline store (Timescale), which stores all data from both online and offline queries                                         |
+| Writes output data to online store database and offline store database                     | Writes output to offline store database and a parquet file containing results to cloud storage. Only writes to online store if specified. |
+
+---
+
+## Running Queries
+
+Queries are executed against deployments or branches. The easiest way to execute
+a query is either: 1). with the Chalk CLI, or 2). with one of Chalk's API
+clients.
+
+### Running Online Queries
+
+Online Queries can be run using the chalk CLI:
+
+```bash
+chalk query --in user.id=1 --in user.name=mary --out user
+```
+
+or one of our API Clients:
+
+```python
+from chalk.client import ChalkClient
+
+client = ChalkClient()
+client.query(
+  inputy={'user.id': 1},
+  output=['user.name'],
+  # branch='test', # run against a branch
+)
+```
+
+### Running Offline Queries
+
+Offline Queries can be run with one of our API Clients.
+
+```python
+from chalk.client import ChalkClient
+from datetime import datetime
+
+client = ChalkClient()
+client.offline_query(
+  input={'user.id': [1,2,3,4]},
+  output=['user.name']
+  # branch='test',                      # run against a branch
+  # recompute_features=True,            # recompute features
+  # run_async=True,                     # run in separate pod from active deployment
+  # max_samples = 10,                   # max of 10 samples
+  # lower_bound=datetime(2024, 10, 12), # sample computed after 10.12.2024
+  # upper_bound=datetime(2024, 10, 20), # sample computed before 10.20.2024
+)
+```
+
+### Scheduled and Triggered Resolver Run
+
+Specific resolvers can also be [scheduled](/docs/resolver-cron) or [triggered](/docs/runs) (for instance as part of engineering pipelines like airflow). Functionally, triggers and schedules can be thought of as queries, which execute on large portions of your data.
```

Comment:
Would drop the comma after "queries". Might be worth exploring the sense in which they can be thought of as queries? Could say something like "queries on large portions of your data that work by pulling inputs from the offline store or external data sources and producing new feature values".